mempool/cat/cache_test.go

```diff
@@ -0,0 +1,97 @@
+package cat
```

Comment:
[non-blocking] to properly test the threadsafe nature of the cache we should add a test that spins up a number of go rountines and goes through the various actions. 

Something like, goroutine A is Adding txns, goroutine B is removing txns, goroutine C is checking and evicting txns. They all have random sleeps in between actions, and then then a common done channel that closes them and the test down after 10s or something.